### PR TITLE
Matching packages between main and develop branches and removal of outdated pull request template

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
The `pytest-subtests` package was added to requirements to enable full testing suite.
The python version in github workflows was rolled back to 3.11.7 to match environments in submodules.
The submodules, `NDSL`, `pyFV3`, and `pySHiELD` are also brought to the latest commits in their respective `develop` branches

**How Has This Been Tested?**
Tested via current test suite

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

**Comment by bensonr** [comment]
@fmalatino - before formally reviewing, wanted to make sure the submodule changes for NDSL, pyFV3, and pySHiELD is expected as part of the rollback to Python 3.11.7

**Comment by fmalatino** [comment]
@bensonr I figured I would bring the `develop` branch up to date with the current versions of the submodules at the same time. I will ammend the initial comment as well.

**Review by oelbert — CHANGES_REQUESTED** [review]
Looks like you'll have to update any calls to constraints.txt now that it's been split into constraints_38 and constaints_311, though it's probably better to just bump the python version to 3.11 and keep one constraint file. If we want to keep support for python 3.8 it's probably better to use environment markers I think?

Also while we're updating the install environment can we delete lines 20-23 from `pace/Dockerfile`? It seems redundant with lines 24-27

**Comment by fmalatino** [comment]
> Looks like you'll have to update any calls to constraints.txt now that it's been split into constraints_38 and constaints_311, though it's probably better to just bump the python version to 3.11 and keep one constraint file. If we want to keep support for python 3.8 it's probably better to use environment markers I think?
> 
> Also while we're updating the install environment can we delete lines 20-23 from `pace/Dockerfile`? It seems redundant with lines 24-27

My vote is that we just move to support for 3.11.7. Can do with the lines in the Dockerfile; will be in next push.

File: Dockerfile
Comment (oelbert) [inline Dockerfile:1]: Thanks for the change. We should probably update the python image in here from 3.8.3 to 3.11.7 as well

**Review by oelbert — CHANGES_REQUESTED** [review]
Great thanks. One more change I think and we should be ok

**Review by oelbert — COMMENTED** [review]
Have you successfully built the container and run a notebook in it?

File: Makefile [hunk 1]
```diff
@@ -87,14 +87,14 @@ endif
 endif
 
 _force_build:
-	DOCKER_BUILDKIT=1 docker build \
+	DOCKER_BUILDKIT=1 podman build \
```
Comment (mlee03) [inline Makefile]: should this be changed to podman?

File: constraints.txt [hunk 2]
```diff
@@ -297,7 +307,7 @@ pathspec==0.12.1
     # via black
 pexpect==4.9.0
     # via ipython
-platformdirs==3.10.0
+platformdirs==2.6.2
```
Comment (mlee03) [inline constraints.txt]: is there a reason to revert patformdirs to 2.5.2?  could we changed platformdirs to 3.10.0?

File: Makefile (hunk 1 above)
Comment (fmalatino) [reply Makefile]: This is a relic from testing on amd, will change back to docker

File: constraints.txt (hunk 2 above)
Comment (fmalatino) [reply constraints.txt]: ~There is no reason, will revert. After re-running `pip-compile --output-file=constraints.txt requirements_dev.txt requirements_docs.txt requirements_lint.txt`, it appears as though `platformdirs==3.10.0` is fine to use though and will be pushed.~ After testing the docker build method, `platformdirs v2.6.2` is necessary; `v3.10.0` created a dependency error.

Comment (mlee03) [reply constraints.txt]: That's funny because the CI container complains with 2.6.2.  Will message you separately about it.

**Review by oelbert — APPROVED** [review]
👍 👍 👍 👍 👍

File: examples/configs/baroclinic_c12.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 stencil_config:
   compilation_config:
-    backend: numpy
+    backend: dace:cpu
```
Comment (bensonr) [inline examples/configs/baroclinic_c12.yaml]: Do we need both numpy and dace:cpu here?  Or do we have other tests for numpy that provide adequate coverage?

Comment (fmalatino) [reply examples/configs/baroclinic_c12.yaml]: Sorry this was a relic from the testing the orchestration issues. Will keep it at numpy only

**Review by bensonr — APPROVED** [review]

**Pull request merged**